src/dispatch/database/service.py

```diff
@@ -355,7 +356,7 @@ def apply_filter_specific_joins(model: Base, filter_spec: dict, query: orm.query
         (SignalInstance, "EntityType"): (SignalInstance.entities, True),
     }
     filters = build_filters(filter_spec)
-    filter_models = get_named_models(filters)[0]
+    filter_models = get_named_models(filters)
```

Comment:
@kevgliss @wssheldon IIRC it was a product of switching to using `SortedSet()` in ` get_named_models()` to respect the order in which we wanted the JOIN statements to happen. We could refactor `get_named_models()` so we don't have to do `[0]`

Reply:
I think that fix is in place, we simply extend the `file_models` instead of creating a nested array (which we had previously just taken the first value of). 

I have tested incidents filtering with the change in place. lmk if you can think other areas that might need to be tested.